### DISCUSSION TITLE
Per group hooks

### DISCUSSION THREAD

**Discussion by mavdotj** [sveltejs/kit#8007] [Ideas]
The problem this would solve is when I have an api group separate from everything else

This is what I originally tried:
```
src/routes/
    (api)/
         api/run/
             +server.ts (doesn't log)
        run/
            +page.svelte (does log, because its a page)
        +layout.server.ts (simply logs that it was run)
```

But this is what I want to happen:
```
src/routes/
    (api)/
         api/run/
             +server.ts (logs "hooks")
        run/
            +page.svelte (logs "layout")
        +hooks.server.ts (logs "hooks")
        +layout.server.ts (logs "layout")
```
So my proposal is a +hooks.sever.ts file

**Comment by mavdotj** [comment]
This is personally a better approach than `event.route.id.startsWith('/(api)/api/')`

**Comment by coryvirok** [comment]
My use case is that I want to add some "expensive" resources to my `event.locals` for `(app)` but not `(marketing)`.

E.g.

```js

export const handle = async ({event, resolve}) => {
  if (event.route.id.startsWith('/(app)/')) {
    event.locals.kafkaClient = await createKafkaClient('...')
    event.locals.statsDClient = await createStatsDClient('...')
    // ...
  }
  const response = await resolve(event)
  return response
}
```

Could also be helpful for not running hooks for routes that need to be pre-rendered but don't need to run `handle()`.

**Comment by mavdotj** [comment]
Working on my own implementation, but it would be easier if #8221 was approved
